bioneuron_oracle/builder.py

```diff
@@ -29,6 +31,22 @@ def __init__(self):
         self.spikes = neuron.h.Vector()
         self.ap_counter.record(neuron.h.ref(self.spikes))
         self.num_spikes_last = 0
+        self._clean = False
+
+    def update(self):
+        if self._clean:
+            raise RuntimeError("cannot update() after cleanup()")
+        count = len(self.spikes) - self.num_spikes_last
+        self.num_spikes_last = len(self.spikes)
+        return count, np.asarray(self.v_record)[-1]
+
+    def cleanup(self):
+        if self._clean:
+            raise RuntimeError("cleanup() may only be called once")
+        self.v_record.play_remove()
+        self.t_record.play_remove()
+        self.spikes.play_remove()
```

Comment:
Note: I haven't actually tested if this solves the memory leak problem you were seeing before. But I'm hoping that this is a better alternative to using `del` for cleanup.